x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -299,7 +299,23 @@ public NamedExpression visitQualifiedNamePattern(EsqlBaseParser.QualifiedNamePat
         // check special wildcard case
         if (patterns.size() == 1) {
             var idCtx = patterns.get(0);
-            if (idCtx.ID_PATTERN().getText().equals(WILDCARD)) {
+            boolean unresolvedStar = false;
+            if (idCtx.ID_PATTERN() != null && idCtx.ID_PATTERN().getText().equals(WILDCARD)) {
+                unresolvedStar = true;
+            }
+            if (idCtx.params() != null) {
+                Expression exp = expression(idCtx.params());
+                if (exp instanceof Literal lit) {
+                    if (lit.value() != null && lit.value().equals(WILDCARD)) {
+                        unresolvedStar = true;
+                    }
+                } else if (exp instanceof UnresolvedAttribute ua) {
+                    if (ua.name() != null && ua.name().equals(WILDCARD)) {
+                        unresolvedStar = true;
+                    }
```

Comment:
When does this scenario occur?  The param being resolved as a literal vs unresolved attribute.

Reply:
I was thinking to tolerate * in constant, identifier and identifier pattern, recognize it as a wildcard no matter how user provides it. However it turns out that the * can be a valid field name by itself, we'd better not tolerating it.
```
curl -u elastic:password -X POST "localhost:9200/existing_index/_doc?pretty" -H 'Content-Type: application/json' -d'
{"*": "Snow Crash", "author": "Neal Stephenson", "release_date": "1992-06-01", "page_count": 470}
'

+ curl -u elastic:password -X POST 'localhost:9200/_sql?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "SELECT * FROM existing_index"
}
'
        *         |     author      |  page_count   |      release_date      
------------------+-----------------+---------------+------------------------
Snow Crash        |Neal Stephenson  |470            |1992-06-01T00:00:00.000Z
Revelation Space  |Alastair Reynolds|585            |2000-03-15T00:00:00.000Z
1984              |George Orwell    |328            |1985-06-01T00:00:00.000Z
Fahrenheit 451    |Ray Bradbury     |227            |1953-10-15T00:00:00.000Z
Brave New World   |Aldous Huxley    |268            |1932-06-01T00:00:00.000Z
The Handmaids Tale|Margaret Atwood  |311            |1985-06-01T00:00:00.000Z

===>keep * adds an extra field *.keyword in the output
+ curl -u elastic:password -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "FROM existing_index | keep * | sort `*`"
}
'
        *         |    *.keyword     |     author      | author.keyword  |  page_count   |      release_date      
------------------+------------------+-----------------+-----------------+---------------+------------------------
1984              |1984              |George Orwell    |George Orwell    |328            |1985-06-01T00:00:00.000Z
Brave New World   |Brave New World   |Aldous Huxley    |Aldous Huxley    |268            |1932-06-01T00:00:00.000Z
Fahrenheit 451    |Fahrenheit 451    |Ray Bradbury     |Ray Bradbury     |227            |1953-10-15T00:00:00.000Z
Revelation Space  |Revelation Space  |Alastair Reynolds|Alastair Reynolds|585            |2000-03-15T00:00:00.000Z
Snow Crash        |Snow Crash        |Neal Stephenson  |Neal Stephenson  |470            |1992-06-01T00:00:00.000Z
The Handmaids Tale|The Handmaids Tale|Margaret Atwood  |Margaret Atwood  |311            |1985-06-01T00:00:00.000Z

+ curl -u elastic:password -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "FROM existing_index | keep `*` | sort `*`"
}
'
        *         
------------------
1984              
Brave New World   
Fahrenheit 451    
Revelation Space  
Snow Crash        
The Handmaids Tale

===> sort wildcard is not allowed
+ curl -u elastic:password -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "FROM existing_index | keep `*` | sort *"
}
'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parsing_exception",
        "reason" : "line 1:39: mismatched input '*' expecting {QUOTED_STRING, INTEGER_LITERAL, DECIMAL_LITERAL, 'false', '(', 'not', 'null', '?', 'true', '+', '-', NAMED_OR_POSITIONAL_PARAM, OPENING_BRACKET, UNQUOTED_IDENTIFIER, QUOTED_IDENTIFIER}"
      }
    ],
    "type" : "parsing_exception",
    "reason" : "line 1:39: mismatched input '*' expecting {QUOTED_STRING, INTEGER_LITERAL, DECIMAL_LITERAL, 'false', '(', 'not', 'null', '?', 'true', '+', '-', NAMED_OR_POSITIONAL_PARAM, OPENING_BRACKET, UNQUOTED_IDENTIFIER, QUOTED_IDENTIFIER}",
    "caused_by" : {
      "type" : "input_mismatch_exception",
      "reason" : null
    }
  },
  "status" : 400
}
```